test/ProductConstructionService.ScenarioTests/ScenarioTests/ScenarioTests_CodeFlowConflicts.cs

```diff
@@ -237,4 +238,170 @@ await CheckIfPullRequestCommentExists(
+
+               // WaitForPullRequestAsync fetches prs in bulk, which doesn't fetch fields like Mergeable and MergeableState which we need
+               pr = await GitHubApi.PullRequest.Get(TestParameters.GitHubTestOrg, TestRepository.VmrTestRepoName, pr.Number);
+               pr.Mergeable.Should().BeFalse();
+               pr.MergeableState.ToString().Should().Be("dirty");
+           });
+    }
+
+    [Test]
+    public async Task Vmr_BackwardConflictFlowTest()
+    {
+        var channelName = GetTestChannelName();
+        var branchName = GetTestBranchName();
+        var productRepo = GetGitHubRepoUrl(TestRepository.TestRepo1Name);
+        var targetBranchName = GetTestBranchName();
+
+        await using AsyncDisposableValue<string> testChannel = await CreateTestChannelAsync(channelName);
+
+        await using AsyncDisposableValue<string> subscriptionId = await CreateBackwardFlowSubscriptionAsync(
+            channelName,
+            TestRepository.VmrTestRepoName,
+            TestRepository.TestRepo1Name,
+            targetBranchName,
+            UpdateFrequency.None.ToString(),
+            TestParameters.GitHubTestOrg,
+            sourceDirectory: TestRepository.TestRepo1Name);
+
+        TemporaryDirectory testRepoFolder = await CloneRepositoryAsync(TestRepository.TestRepo1Name);
+        TemporaryDirectory vmrFolder = await CloneRepositoryAsync(TestRepository.VmrTestRepoName);
+        var newFileInVmrPath = Path.Combine(vmrFolder.Directory, "src", TestRepository.TestRepo1Name, TestFile1Name);
+        var newFilePath = Path.Combine(testRepoFolder.Directory, TestFile1Name);
+
+        await CreateTargetBranchAndExecuteTest(targetBranchName, testRepoFolder.Directory, async () =>
+        {
+            using (ChangeDirectory(vmrFolder.Directory))
+            {
+                await using (await CheckoutBranchAsync(branchName))
+                {
+                    // Make a change in the VMR
+                    TestContext.WriteLine("Making code changes in the VMR");
+                    File.WriteAllText(newFileInVmrPath, "not important");
+
+                    await GitAddAllAsync();
+                    await GitCommitAsync("Add new file");
+
+                    // Push it to github
+                    await using (await PushGitBranchAsync("origin", branchName))
+                    {
+                        var repoSha = (await GitGetCurrentSha()).TrimEnd();
+
+                        // Create a new build from the commit and add it to a channel
+                        Build build = await CreateBuildAsync(
+                            GetGitHubRepoUrl(TestRepository.VmrTestRepoName),
+                            branchName,
+                            repoSha,
+                            "1",
+                            // We might want to add some assets here to mimic what happens in the VMR
+                            []);
```

Comment:
We might want to add some assets that will be in the first build only and not in the second to see if those still show up in the PR? Otherwise we're not really testing anything?

Reply:
I copied this setup from the backflow test and didn't notice the comment, but I can add some.

Reply:
But honestly I'm not sure if this would test anything specific to the conflict test? It'd just be equivalent to two subsequent backflow builds, which should be a common scenario